src/wmtk/io/MshReader.cpp

```diff
@@ -0,0 +1,316 @@
+#include "MshReader.hpp"
+
+#include <wmtk/EdgeMesh.hpp>
+#include <wmtk/PointMesh.hpp>
+#include <wmtk/TetMesh.hpp>
+#include <wmtk/TriMesh.hpp>
+
+#include <wmtk/utils/mesh_utils.hpp>
+
+namespace wmtk {
+
+
+std::shared_ptr<Mesh> MshReader::read_aux(const std::filesystem::path& filename)
+{
+    m_spec = mshio::load_msh(filename.string());
+
+    std::shared_ptr<Mesh> res;
+
+    if (get_num_tets() > 0) {
+        V.resize(get_num_tet_vertices(), 3);
+        S.resize(get_num_tets(), 4);
+
+        extract_tet_vertices();
+        extract_tets();
+
+        auto tmp = std::make_shared<TetMesh>();
+        tmp->initialize(S);
+        res = tmp;
+    } else if (get_num_faces() > 0) {
+        V.resize(get_num_face_vertices(), 3);
+        S.resize(get_num_faces(), 3);
+
+        extract_face_vertices();
+        extract_faces();
+
+        auto tmp = std::make_shared<TriMesh>();
+        tmp->initialize(S);
+        res = tmp;
+    } else if (get_num_edges() > 0) {
+        V.resize(get_num_edge_vertices(), 3);
+        S.resize(get_num_edges(), 2);
+
+        extract_edge_vertices();
+        extract_edges();
+
+        auto tmp = std::make_shared<EdgeMesh>();
+        tmp->initialize(S);
+        res = tmp;
+    } else {
+        res = std::make_shared<PointMesh>();
+    }
+
+
+    mesh_utils::set_matrix_attribute(V, "vertices", PrimitiveType::Vertex, *res);
+
+    return res;
+}
+
+
+void MshReader::extract_tet_vertices()
+{
+    return extract_vertices<3>();
+}
+
+void MshReader::extract_face_vertices()
+{
+    return extract_vertices<2>();
+}
+
+void MshReader::extract_edge_vertices()
+{
+    extract_vertices<1>();
+}
+
+void MshReader::extract_edges()
+{
+    extract_simplex_elements<1>();
+}
+
+void MshReader::extract_faces()
+{
+    extract_simplex_elements<2>();
+}
+
+void MshReader::extract_tets()
+{
+    extract_simplex_elements<3>();
+}
+
+// std::vector<std::string> MshReader::get_edge_vertex_attribute_names() const
```

Comment:
Is that dead code or will it ever be implemented? Either add a TODO comment or remove it.